### ISSUE TITLE
Expose State Channels explicitly

### ISSUE THREAD

**Issue by thepiwo** [aeternity/ae_mdw#823]
we should have an api that returns state channel information per channel aggregated

![image](https://user-images.githubusercontent.com/6539235/182773283-cf2c17fa-307b-41da-9366-29a0c66bde45.png)

(@janmichek )

**Comment by marc0olo** [comment]
@sborrazas @jyeshe @janmichek @thepiwo 

we should make sure that we can serve the data of following two screens. if mdw shouldn't be responsible for everything displayed there we have to find alternative solutions to display accordingly:

# State Channels

## General View
![image](https://user-images.githubusercontent.com/8124114/186772656-4d117944-6ab2-4a67-93cf-172da6d08af0.png)

## Detail View
![image](https://user-images.githubusercontent.com/8124114/186772831-2013bb11-d701-4d6f-b819-6c1496d23d62.png)

**Comment by janmichek** [comment]
Thanks for summing things up.

Just to clarify it...

Data requested AT this issue is meant to be displayed them on the dashboard.
https://aeternity-blockchain.atlassian.net/browse/EX-75


Data grouped by time meant to be display on State Channels subpage that will be implemented later (not for the first release)
https://aeternity-blockchain.atlassian.net/browse/EX-121

**Comment by marc0olo** [comment]
> Data grouped by time meant to be display on State Channels subpage that will be implemented later (not for the first release)
> https://aeternity-blockchain.atlassian.net/browse/EX-121

yes, I just wanted to let you know that I think it'd be cool to focus on the state channel area as soon as possible as we aim to release the state channel demo in the near future :-)

**Comment by janmichek** [comment]
No probs... matter of prioritization

**Comment by thepiwo** [comment]
@sborrazas can you check?

**Comment by sborrazas** [comment]
The two things that aren't directly accessible via the mdw API are:
1. Accounts with most open/active channels
2. The "Balanced locked over time", which can only be accessed by getting the full list of channel txs for that channel right now

For both of these features we would need to add an additional index and endpoint. Since there isn't many txs per chanel right now we can solve 2. by simply accessing the list of channel txs for a specific channel. 

As for 1., how does an account have more than 1 "open channel"? Would this data be obtainable by looking at the few "OpenChannelTx" from the /txs endpoint as well?

**Comment by marc0olo** [comment]
> Since there isn't many txs per chanel right now we can solve 2. by simply accessing the list of channel txs for a specific channel.

in general I don't expect many on-chain actions per channel.


> As for 1., how does an account have more than 1 "open channel"? Would this data be obtainable by looking at the few "OpenChannelTx" from the /txs endpoint as well?

yes, as long there is no on-chain tx to close a specific channel the channel is kept open and an account can have many different open channel with different counterparties

**Comment by sborrazas** [comment]
@marc0olo what does it mean for an account to have a channel? That the account is an "initiator" for that channel? Or a responder as well?

As for the general view, I realized we're going to need 1 more endpoint as well to describe the full list of open channels

**Comment by marc0olo** [comment]
> @marc0olo what does it mean for an account to have a channel? That the account is an "initiator" for that channel? Or a responder as well?

`OpenChannelTx` is only valid if initiator and responder provide a valid signature. I will let @velzevur and @Kalovelo confirm this.

so from my interpretation it'd be valid for both, responder and initiator, as they agree on "opening" a channel ;-)
